android/src/main/java/ly/count/dart/countly_flutter/CountlyFlutterPlugin.java

```diff
@@ -214,19 +225,29 @@ public void onComplete(Task<InstanceIdResult> task) {
                   pushTokenType = Countly.CountlyMessagingMode.PRODUCTION;
               }
               result.success("pushTokenType!");
+          } else if ("registerForNotification".equals(call.method)) {
+              registerForNotification(args, new Callback() {
+                  @Override
+                  public void callback(final String resultString) {
+                      activity.runOnUiThread(new Runnable() {
+                          @Override
+                          public void run() {
+                              result.success(resultString);
```

Comment:
Why is this returned inside of the callback?

Reply:
This callback is stored and when you receive notification, this callback is triggered.

Reply:
what does this line do?
"result.success(resultString);"

Reply:
That's push data in string format.